src/main/java/org/jenkinsci/plugins/stashNotifier/StashNotifier.java

```diff
@@ -226,14 +236,20 @@ private boolean perform(Run<?, ?> run,
 
 		PrintStream logger = listener.getLogger();
 
-		Result result = run.getResult();
-		if (result == null && disableInProgress) {
+		Result buildResult = run.getResult();
+		if (buildResult == null && disableInProgress) {
 			return true;
-		} else if (result == null) {
+		} else if (buildResult == null) {
 			state = StashBuildState.INPROGRESS;
-		} else if (result.equals(Result.SUCCESS)) {
+		} else if (buildResult == Result.SUCCESS) {
 			state = StashBuildState.SUCCESSFUL;
-		} else if (result.equals(Result.NOT_BUILT)) {
+		} else if (buildResult == Result.UNSTABLE && considerUnstableAsSuccess) {
+			logger.println("UNSTABLE reported to stash as SUCCESSFUL");
+			state = StashBuildState.SUCCESSFUL;
+		} else if (buildResult == Result.ABORTED && (disableAbortAsFailures || disableInProgress)) {
```

Comment:
Should this be `(disableAbortAsFailures && disableInProgress)` ? We should not allow the notifier to leave the build blue, so if we have not disable inprogress thus we cannot drop aborted without marking?